libs/wingsdk/src/simulator/simulator.ts

```diff
@@ -867,25 +919,36 @@ export class Simulator {
 
     // initialize the resource with the simulator context
     const context = this.createContext(resourceConfig, handle);
-    const attrs = await resourceObject.init(context);
+    let initError: any;
+    try {
+      const attrs = await resourceObject.init(context);
+
+      this.state[path].attrs = {
+        ...this.state[path].attrs,
+        ...attrs,
+      };
+    } catch (error: any) {
+      initError = error;
+    }
```

Comment:
I think there are other parts of the simulator expecting to catch and handle initialization errors - I wonder if they need to be updated somehow somehow to be compatible with this change

https://github.com/winglang/wing/blob/cdfab81a04acc494f84d01b74838e8cba72fb964/libs/wingsdk/src/simulator/simulator.ts#L279-L309

Reply:
What I've seen is that constructs (e.g. resource.inflight.ts) swallowed their errors, emitted an error trace and returned an empty attrs object.

If we want to shut everything down if a construct fails to initialize, the UX may be impacted (e.g. no bucket can be inspected while any construct failed).